### ISSUE TITLE
Renaming datasets eliminated re-run button

### ISSUE THREAD

**Issue by nekrut** [galaxyproject/galaxy#16868]
If you rename a dataset th ererun button and preview disapper. Here is an example:

## Before:

![image](https://github.com/galaxyproject/galaxy/assets/4291636/2f6383ea-1955-45b4-99a4-ce19db334a1d)

## After:

(This has different dataset number because I screwed up, but the idea is the same)

![image](https://github.com/galaxyproject/galaxy/assets/4291636/d02d3ceb-0b26-4362-b8f7-f627b6b9510c)

**Comment by itisAliRH** [comment]
I could reproduce it on database change action (not rename):


https://github.com/galaxyproject/galaxy/assets/8046843/be528e5e-e2d1-4677-a685-570b84398f64

**Comment by ahmedhamidawan** [comment]
I've been trying to reproduce this in multiple ways but I've had no success... 
The logic for showing the rerun button is sound as well so the only way this could happen is if the following properties changed:
```
showRerun() {
    return (
        this.item.rerunnable &&
        this.item.creating_job &&
        this.item.state != "upload" &&
        this.item.state != "noPermission"
    );
},
```

**Comment by ahmedhamidawan** [comment]
Fixed in #17448
